src/concurrency/sync.rs

```diff
@@ -610,14 +630,14 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     fn condvar_wait(
         &mut self,
         condvar: CondvarId,
-        mutex: MutexId,
+        mutex_ref: &MutexRef,
```

Comment:
This one needs to clone anyway so it makes sense that it would take an owned ref (so we can avoid double-cloning).